src/DashJS/SDK/Platform/methods/documents/broadcast.ts

```diff
@@ -1,20 +1,32 @@
-import {Platform} from "../../Platform";
-
+// this file is deprecated
+import broadcast = require('../record/broadcast');
 
+<<<<<<< HEAD
+/**
+ * Broadcast document onto the platform
+ * 
+ * @param {Platform} this - bound instance class
+ * @param document - document
+ * @param identity - identity
+ */
 export async function broadcast(this: Platform, document: any, identity: any): Promise<any> {
-    const { account, client, dpp } = this;
+    const {account, client, dpp} = this;
 
     // @ts-ignore
     const identityHDPrivateKey = account.getIdentityHDKey(0, 'user');
 
     // @ts-ignore
     const identityPrivateKey = identityHDPrivateKey.privateKey;
 
-    const stateTransition = dpp.document.createStateTransition([document]);
+    const stateTransition = dpp.dataContract.createStateTransition([document]);
```

Comment:
This one might be problematic. 
DocumentStateTransition and DataContractStateTransition are both extending AbstractStateTransition with some minor changes (right now only an ID). 
However, we need to keep it this way. 
May be, by knowing record type (dataContract or document) we could just do a `dpp[recordType].createStateTransition` as a record.broadcast logic.

Reply:
Please refer to [records/broadcast.ts](src/DashJS/SDK/Platform/methods/record/broadcast.ts)

Reply:
Yeah, I should have commented there, but I do refer to the same issue, where records/broadcast.ts refers to `dataContract` as well as in the case of records of type documents.

Reply:
>The small minor changes is on which data type factory we are using.
Which would be fixed I guess easily via a dpp[recordType] checkup (if recordType is an available factory within dpp)

Yes, I see. Well, leave it as it was then. 

Reverted

Reply:
I will switch from merging your PR to master to a branch locally, this way I will contribute on that part. I kinda like the idea of a Record of type dataContract or document. 
Thanks for having reverted in the mean time :)